Core/src/Managers/AudioManager.cpp

```diff
@@ -1,15 +1,29 @@
-﻿#include "Managers/AudioManager.h"
-#include <iostream>
+﻿#include <Managers/AudioManager.h>
+#include <Components/AudioComponent.h>
+#include <Managers/EntityManager.h>
+#include <Components/ListenerComponent.h>
+
+AudioManager* AudioManager::pInstance;
+std::mutex AudioManager::mutex_;
+
+AudioManager* AudioManager::GetInstance() {
+    if (pInstance == nullptr) {
```

Comment:
If I remember correctly, it is making sure that pInstance isn't instanced between passing the check and grabbing the mutex. If perchance something instances the manager and then releases the lockguard, this prevents reinitialization of it. In theory it is fine, but its a secondary protection that is low impact as getting the instance is not called frequently.

Reply:
Then why not put the mutex above the conditional. 

Because you put the temp variable in, yes there's a potential for a reinitialization, but you already have a static pInstance, that does not reinitialize if you set it using the next comment.

Reply:
look at bornas slide 
![image](https://github.com/user-attachments/assets/dabf082c-b283-4642-a159-64742fa0329d)